repository_service_tuf_worker/repository.py

```diff
@@ -853,7 +854,73 @@ def bump_online_roles(self) -> bool:
                     "timeout."
                 )
                 raise redis.exceptions.LockError(
-                    f"RSTUF: Task exceed `LOCK_TIMEOUT` ({timeout} seconds)"
+                    f"RSTUF: Task exceed `LOCK_TIMEOUT` ({self._timeout} "
+                    "seconds)"
                 )
 
         return True
+
+    def metadata_rotation(
+        self,
+        payload: Dict[str, Any],
+        update_state: Optional[
+            Task.update_state
+        ] = None,  # It is required (see: app.py)
+    ) -> Dict[str, Any]:
+        metadata = payload.get("metadata")
+        if metadata is None:
+            raise KeyError("No 'metadata' in the payload")
+
+        new_root: Metadata[Root] = Metadata.from_dict(metadata.get(Root.type))
+        old_root: Metadata[Root] = self._storage_backend.get(Root.type)
+        old_root_online_key = old_root.signed.roles[Timestamp.type].keyids
+        new_root_online_key = new_root.signed.roles[Timestamp.type].keyids
+
+        if old_root.signed.version + 1 != new_root.signed.version:
+            raise BadVersionNumberError(
+                f"New root version not expected {new_root.signed.version}"
+            )
+
+        # We always persist the new root metadata, but we cannot persist
+        # without verifying if the online key is rotate to avoid mismatch with
+        # the online public key.
+        if old_root_online_key == new_root_online_key:
+            # online key is not changed, persist the new root
+            self._persist(new_root, Root.type)
+            logging.info(f"Updating root metadata: {new_root.signed.version}")
+
+        else:
+            # we lock this action to stop all new publish targets
+            # (`publish_targets`) to avoid incosistence to the TUF clients
```

Comment:
```suggestion
            # We lock this action to stop all new tasks that publish targets
            # (`publish_targets`) to avoid inconsistencies happening to the
            # TUF clients
```